File: src/main/java/ru/nucodelabs/gem/app/io/SonetImportManager.java
package ru.nucodelabs.gem.app.io;

import ru.nucodelabs.data.ves.Picket;

import java.io.File;

public interface SonetImportManager {

    static SonetImportManager create() {
        return new SonetImportManagerImpl();
    }

    /**
     * Загружает имя и экспериментальные данные в пикет
     */
    Picket loadNameAndExperimentalDataFromEXPFile(File expFile, Picket target) throws Exception;

    /**
     * Загружает только экспериментальные данные в пикет
     */
    Picket loadExperimentalDataFromEXPFile(File expFile, Picket target) throws Exception;

    /**
     * Загружает модельные данные в пикет
     */
    Picket loadModelDataFromMODFile(File modFile, Picket target) throws Exception;
}

File: src/main/java/ru/nucodelabs/gem/app/io/SonetImportManager.kt
package ru.nucodelabs.gem.app.io

import ru.nucodelabs.geo.ves.ModelLayer
import ru.nucodelabs.geo.ves.Picket
import java.io.File

interface SonetImportManager {
    /**
     * Загружает имя и экспериментальные данные в пикет
     */
    @Throws(Exception::class)
    fun fromEXPFile(expFile: File): Picket

    /**
     * Загружает данные модели
     */
    @Throws(Exception::class)
    fun fromMODFile(modFile: File): List<ModelLayer>

    companion object Factory {
        @JvmStatic
        fun create(): SonetImportManager = SonetImportManagerImpl()
    }
}